x/ccv/provider/keeper/relay.go

```diff
@@ -132,34 +167,27 @@ func (k Keeper) EndBlockCallback(ctx sdk.Context) {
 	k.IncrementValidatorSetUpdateId(ctx)
 }
 
-// OnRecvPacket slashes and jails the given validator in the packet data
-func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data ccv.SlashPacketData) exported.Acknowledgement {
+// OnRecvSlashPacket slashes and jails the given validator in the packet data
+func (k Keeper) OnRecvSlashPacket(ctx sdk.Context, packet channeltypes.Packet, data ccv.SlashPacketData) exported.Acknowledgement {
 	// check that the channel is established
-	chainID, ok := k.GetChannelToChain(ctx, packet.DestinationChannel)
-	if !ok {
-		ack := channeltypes.NewErrorAcknowledgement(
-			sdkerrors.Wrap(
-				channeltypes.ErrInvalidChannelState,
-				packet.DestinationChannel,
-			).Error(),
-		)
-		chanCap, _ := k.scopedKeeper.GetCapability(ctx, host.ChannelCapabilityPath(packet.DestinationPort, packet.DestinationChannel))
-		k.channelKeeper.ChanCloseInit(ctx, packet.DestinationPort, packet.DestinationChannel, chanCap)
-		return &ack
+	chainID, found := k.GetChannelToChain(ctx, packet.DestinationChannel)
+	if !found {
+		// Slash packet was sent on a channel different than any of the established CCV channels
+		return utils.OnRecvPacketOnUnknownChannel(ctx, k.scopedKeeper, k.channelKeeper, packet)
 	}
 
 	// apply slashing
-	if err := k.HandleSlashPacket(ctx, chainID, data); err != nil {
-		ack := channeltypes.NewErrorAcknowledgement(err.Error())
-		return &ack
+	if _, err := k.HandleSlashPacket(ctx, chainID, data); err != nil {
+		errAck := channeltypes.NewErrorAcknowledgement(err.Error())
+		return &errAck
 	}
 
 	ack := channeltypes.NewResultAcknowledgement([]byte{byte(1)})
 	return ack
 }
 
-// HandleSlashPacket slash and jail a wrong doing validator according the infraction height and type
-func (k Keeper) HandleSlashPacket(ctx sdk.Context, chainID string, data ccv.SlashPacketData) error {
+// HandleSlashPacket slash and jail a misbehaving validator according the infraction type
+func (k Keeper) HandleSlashPacket(ctx sdk.Context, chainID string, data ccv.SlashPacketData) (success bool, err error) {
```

Comment:
where does the success bool get used if its ignored above?

Reply:
In tests. For example, there was a test that expected `HandleSlashPacket` to return error if the validator is tombstoned. However, in case of error while executing `HandleSlashPacket()`, `OnRecvSlashPacket()` returns an error ack and if the validator is already tombstoned, there should be no error ack. i.e., https://github.com/cosmos/interchain-security/blob/345fed1ffe11b27c74ef1c5dbbdf1969e75ac0da/x/ccv/provider/keeper/keeper_test.go#L405